pkg/router/controller/status.go

```diff
@@ -167,16 +183,22 @@ func performIngressConditionUpdate(action string, lease writerlease.Lease, track
 	})
 }
 
-// recordIngressCondition updates the matching ingress on the route (or adds a new one) with the specified
+// recordIngressCondition adds, updates, or removes the matching ingress on the route with the specified
 // condition, returning whether the route was updated or created, the time assigned to the condition, and
 // a pointer to the current ingress record.
-func recordIngressCondition(route *routev1.Route, name, hostName string, condition routev1.RouteIngressCondition) (changed, created bool, at time.Time, latest, original *routev1.RouteIngress) {
+func recordIngressCondition(route *routev1.Route, name, hostName string, condition *routev1.RouteIngressCondition) (changed, created bool, at time.Time, latest, original *routev1.RouteIngress) {
 	for i := range route.Status.Ingress {
 		existing := &route.Status.Ingress[i]
 		if existing.RouterName != name {
 			continue
 		}
 
+		// If condition is nil, remove the ingress condition.
```

Comment:
This should be targeting a specific condition, not removing the entire ingress condition, unless perhaps there isn't any conditions left.

Reply:
I'll take that back. When a router unadmits a route, we should just clear the whole status.ingress object. So this is mostly fine.